sdk/identity/identity/Troubleshooting.md

```diff
@@ -0,0 +1,241 @@
+
+When you get the error `The request body must contain the following parameter: 'client_assertion' or 'client_secret'`, it happens because of how the AAD application is configured. The AAD app registration seems to be configured as a confidential application. The `UsernamePassword` credential works only with public clients and doesn't support confidential applications. If you need to support confidential applications, use `ClientSecret` or `ClientCertificate` Credentials instead.
+
+In order to allow public client authentication on your AAD, follow these steps:
+
+- Go to the Azure Portal.
+- Go to the `Authentication` page.
+- Scroll to the bottom of this page, and you will see something that says: `Allow public client flows`, and close to that, you will see a `yes` and `no` switch. Make sure this is set to `yes`.
+  After that, you should not need to specify a client secret to authenticate with this credential.
+
+## Troubleshooting Service Principal Authentication Issues
+
+### Invalid Argument Issues
+
+#### Client Id
+
+The Client Id is the application Id of the registered application / service principal in Azure Active Directory.
+It is a required parameter for `ClientSecretCredential` and `ClientCertificateCredential`. If you have already created your service principal
+then you can retrieve the client/app id by following the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
+
+#### Tenant Id
+
+The tenant id is the Global Unique Identifier (GUID) that identifies your organization. It is a required parameter for
+`ClientSecretCredential` and `ClientCertificateCredential`. If you have already created your service principal
+then you can retrieve the client/app id by following the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
+
+### Client Secret Credential Issues
+
+#### Client Secret Argument
+
+The client secret is the secret string that the application uses to prove its identity when requesting a token, this can also can be referred to as an application password.
+If you have already created a servie principal you can follow the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret) to get the client secret for your application.
+
+### Client Certificate Credential Issues
+
+#### Client Certificate Argument
+
+The `Client Certificate Credential` accepts `pfx` and `pem` certificates. The certificate needs to be associated with your registered application/service principal. To create and associate a certificate with your registered app. please follow the instrucitons [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#option-1-upload-a-certificate).
```

Comment:
In our case, we only accept `pem` certificates, no `pfx` at the moment.

Reply:
Thanks!